packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -502,6 +512,87 @@ type GenericPrefetchThunk = (
   options: PrefetchOptions
 ) => ThunkAction<void, any, any, AnyAction>
 
+/**
+ * @internal
+ */
+type CreateSuspendableQueryOptions<
+  Definitions extends EndpointDefinitions,
+  Key extends keyof Definitions
+> = {
+  name: Key
+  isSkipped: boolean
+  args: any
+  api: Api<any, Definitions, any, any, CoreModule>
+  prefetch: (arg: any, options?: PrefetchOptions | undefined) => void
+  queryStateResults: UseQueryStateResult<
+    QueryDefinitionOf<Key, Definitions>,
+    UseQueryStateDefaultResult<QueryDefinitionOf<Key, Definitions>>
+  >
+}
+
+type QueryDefinitionOf<
+  Key extends keyof Definitions,
+  Definitions extends EndpointDefinitions
+> = Definitions[Key] extends QueryDefinition<infer A, infer B, infer C, infer D>
+  ? QueryDefinition<A, B, C, D>
+  : never
+
+const createSuspendablePromise = <
+  Definitions extends EndpointDefinitions,
+  Key extends QueryKeys<Definitions>
+>({
+  isSkipped,
+  args,
+  prefetch,
+  api,
+  name,
+  queryStateResults,
+}: CreateSuspendableQueryOptions<
+  Definitions,
+  Key
+>): Suspendable['getSuspendablePromise'] => {
+  const retry = () => {
+    prefetch(args, {
+      force: true,
+    })
+  }
+
+  return (): Promise<unknown> | undefined => {
+    // We do not suspend if a query is skipped:
+    // @see https://github.com/vercel/swr/pull/357#issuecomment-627089889
+    if (!isSkipped && typeof queryStateResults.data === 'undefined') {
+      if (queryStateResults.isLoading) {
+        let pendingPromise = api.util.getRunningOperationPromise(name, args)
+
+        if (!pendingPromise) {
+          prefetch(args, {
+            force: true,
+          })
```

Comment:
```suggestion
          retry()
```
or am I missing something?